packages/components/radio-group/src/radio-group.js

```diff
@@ -0,0 +1,145 @@
+import { TSElement, unsafeCSS, html, customElementDefineHelper } from '@tradeshift/elements';
+import css from './radio-group.css';
+import { getDirectionByKey, customEventNames } from './utils';
+
+import '@tradeshift/elements.radio';
+
+customElementDefineHelper(
+	'ts-radio-group',
+	class extends TSElement {
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				dir: { type: String, reflect: true },
+				focused: { type: Boolean, reflect: true },
+				index: { type: Number, reflect: true },
+				value: { type: String, reflect: true },
+				title: { type: String, reflect: true }
+			};
+		}
+
+		get direction() {
+			return this.dir ? this.dir : this.bodyDir;
+		}
+
+		toggle(radioNode) {
+			if (!radioNode || radioNode.disabled) {
+				return;
+			}
+
+			for (const radio of this.radios) {
+				radio.checked = radio === radioNode;
+				if (radio.checked) {
+					const index = this.radios.indexOf(radio);
+					const value = radio.value;
+					this.index = index;
+					this.value = value;
+					this.dispatchCustomEvent(customEventNames.RADIO_SELECTED, { radio, index, value });
+				}
+			}
+		}
+
+		firstUpdated() {
+			this.shadowRoot.host.addEventListener(customEventNames.RADIO_CLICK, this.handleRadioClick);
+			this.shadowRoot.host.addEventListener(customEventNames.RADIO_FOCUS, this.handleFocus);
+			this.shadowRoot.host.addEventListener(customEventNames.RADIO_BLUR, this.handleBlur);
+			this.handleSlotChange();
+		}
+
+		disconnectedCallback() {
+			this.shadowRoot.host.removeEventListener(customEventNames.RADIO_CLICK, this.handleRadioClick);
+			this.shadowRoot.host.removeEventListener(customEventNames.RADIO_FOCUS, this.handleFocus);
+			this.shadowRoot.host.removeEventListener(customEventNames.RADIO_BLUR, this.handleBlur);
+		}
+
+		handleRadioClick(e) {
+			e.stopPropagation();
+			this.toggle(e.target);
+		}
+
+		handleFocus(e) {
+			e.stopPropagation();
+			this.focused = true;
+		}
+
+		handleBlur(e) {
+			e.stopPropagation();
+			this.focused = false;
+		}
+
+		handleSlotChange() {
+			this.radios = Array.from(this.shadowRoot.host.querySelectorAll('ts-radio'));
+
+			for (let i = 0; i < this.radios.length; i++) {
+				const radio = this.radios[i];
+				if (radio.checked && !radio.disabled) {
+					this.index = i;
+				}
+			}
+			if (this.index) {
+				this.toggle(this.radios[this.index]);
+			}
+		}
+
+		handleKeyDown(e) {
+			if (this.radios.length === 0) {
+				return;
+			}
+
+			const enabledRadios = this.radios.filter(checkbox => !checkbox.disabled);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "checkbox" hides or potentially hides a variable declared in an outer scope at line 102. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AVariableShadowing)